docs/hackathons/ihp_10_2022.rst

```diff
@@ -0,0 +1,285 @@
+
+
+Moreover, we force each space to contain only one metric, since an equipped instance of a space represents :math:`(M, g)`.
+
+
+
+Similar actions will be done regarding equipping a space with a group action.
+
+
+
+Intrinsic vs extrinsic coordinates (manifold)
+*********************************************
+
+At a lower level, a manifold is expected to implement points via ``extrinsic`` OR ``intrinsic`` coordinates, never both.   
+
+
+This means that code as the following will not be allowed:
+
+
+.. code-block:: python
+
+    space = MySpace(default_coords_type="extrinsic")
+
+    space.default_coords_type = "intrinsic"
+
+
+That is, instances of a manifold are expected to be static, i.e. if you want a different point representation of your space, you should instantiate a new object.
+
+
+The main rationale behind this decision is maintainability. For example, each method of the manifold knows exactly which array shape to expect (and what each array represents). This will remove a lot of ifs from the code base (remember that ifs relate to complexity). It will also allow to have a more meaningful code hierarchy, as different representations of the same space may rely on different definitions of manifold.
+
+
+Besides, we expect usability to increase, as a user also knows exactly which kind of points each of the objects is able to manipulate.
+
+
+For the end user, no significant change will happen at instantiation level, as we will rely on interfaces (similarly to what is currently done with ``Hyperbolic``). An interface in this case is simply a class with ``__new__`` which returns an instance of another class. For example:
+
+
+.. code-block:: python
+
+    class Hypersphere:
+
+        def __new__(self, dim, intrinsic=False, equip=True):
+            if intrinsic:
+                return IntrinsicHypersphere(dim, equip=equip)
+            else:
+                return ExtrinsicHypersphere(sim, equip=equip) 
+
+
+
+Intrinsic vs extrinsic coordinates (connection)
+***********************************************
+
+Similarly to the manifold, the connection inheritance structure will also be affected by the clearer distinction between ``extrinsic`` and ``intrinsic`` (in fact, changes in the manifold structure tend to be reflected in the connection structure and vice-versa).
+
+
+Here, this distinction will reduce namespace pollution ("ill-defined" methods will not be available to many children), make testing easier (e.g. currently there are many skips in tests due to "inappropriate" method inheritance), and reduce information overload to a class user (e.g. ``dir(<MyMetric>)`` will only show methods that are properly defined).
+
+
+To make it more concrete, we will keep in ``Connection`` only methods that return coordinate invariant objects (e.g. scalars), and move coordinate dependent methods to ``IntrinsicConnection`` (e.g. ``christoffels``, ``riemannian_tensor``, etc).
```

Comment:
What actually remains in the extrinsic case?

Reply:
actually the current implementations of exp/log in connection are for the intrinsic case, and I don't think we have an extrinsic case in the code base.